map/bookmark_helpers.cpp

```diff
@@ -370,33 +371,40 @@ std::string GetKMLorGPXPath(std::string const & filePath)
     if (kmlData == nullptr)
       return {};
 
-    fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
+    auto fileSavePath = GenerateValidAndUniqueFilePathForKML(base::FileNameFromFullPath(filePath));
     if (!SaveKmlFileByExt(*kmlData, fileSavePath))
       return {};
+    return {std::move(fileSavePath)};
   }
   else if (fileExt == kKmzExtension)
   {
-    // Extract KML file from KMZ archive and save to temp KML with unique name.
+    // Extract KML files from KMZ archive and save to temp KMLs with unique name.
+    std::vector<std::string> kmlFilePaths;
     try
     {
       ZipFileReader::FileList files;
       ZipFileReader::FilesList(filePath, files);
-      std::string kmlFileName;
-      std::string ext;
-      for (size_t i = 0; i < files.size(); ++i)
+      std::string indexFileName;
+      std::vector<std::string> kmlFileNames;
+      for (const auto & file : files)
       {
-        ext = GetLowercaseFileExt(files[i].first);
+        auto const & kmlFileName = file.first;
+        std::string ext = GetLowercaseFileExt(kmlFileName);
         if (ext == kKmlExtension)
-        {
-          kmlFileName = files[i].first;
-          break;
-        }
+          kmlFileNames.push_back(kmlFileName);
       }
-      if (kmlFileName.empty())
-        return {};
-
-      fileSavePath = GenerateValidAndUniqueFilePathForKML(kmlFileName);
-      ZipFileReader::UnzipFile(filePath, kmlFileName, fileSavePath);
+      for (const auto & kmlFileName : kmlFileNames)
+      {
+        auto fileSavePath = GenerateValidAndUniqueFilePathForKML(kmlFileName);
+        ZipFileReader::UnzipFile(filePath, kmlFileName, fileSavePath);
+        auto start = kmlFileName.find_last_of('/') + 1;
+        auto const name = kmlFileName.substr(start, kmlFileName.size() - start);
+        // skip index kml if there are more than 1 kml files in kmz
+        if (name == "doc.kml" && kmlFileNames.size() > 1)
```

Comment:
In theory we can have both Placemark and NetworkLink inside document according to the schema: https://developers.google.com/kml/documentation/kmlreference#syntax_8. But what is the correct way to handle such cases, if we want to? 
Looks like we should skip, if file has only index. But if we use our kml parser we will just have an empty FileData - should we skip all FileData objects if there are no bookmarks and tracks there?

Reply:
Good questions. I don't see any reason to import empty bookmark lists. It would be great to have this check in the right place.

Let's _assume_ that root kml won't have any useful info and skip it, a comment in the code can be added.

Reply:
Comment is added